strawberry/field.py

```diff
@@ -134,8 +135,20 @@ def __init__(
         self.is_subscription = is_subscription
 
         self.permission_classes: List[Type[BasePermission]] = list(permission_classes)
-        self.directives = directives
-
+        self.directives = list(directives)
+        self.extensions: Optional[List[FieldExtension]] = extensions
+
+        # Automatically add the permissions extension
+        if len(self.permission_classes):
+            from .permission import PermissionExtension
+
+            if not self.extensions:
+                self.extensions = []
+            permission_instances = [
+                permission_class() for permission_class in permission_classes
+            ]
+            # Append to make it run first (last is outermost)
+            self.extensions.append(PermissionExtension(permission_instances))
```

Comment:
We should decide wether we want to deprecate `permission_classes` as a way of adding Permissions to a field in favor of just using the `PermissionsExtension`. I think this is a duplicate way to do things and may lead to confusion for new users. Just using the extension is more straight forward, as it has no side-effects on extensions and enables + encourages customization as opposed to this piece of 'magic'.